### ISSUE TITLE
Locale does not apply to dates created in util.js

### ISSUE THREAD

**Issue by tommedema** [clauderic/react-infinite-calendar#43]
When applying locale like so:

```
const locale = {
            name: 'nl',
            todayLabel: {
                long: 'Vandaag'
            },
            week: {
                dow: 0 // week starts on sunday
            },
            months : "januari_februari_maart_april_mei_juni_juli_testing_september_oktober_november_december".split("_"),
            monthsShort : "jan_feb_mar_apr_mei_jun_jul_tst_sep_okt_nov_dec".split("_"),
            weekdays : "Zondag_Maandag_Dinsdag_Woensdag_Donderdag_Vrijdag_Zaterdag".split("_"),
            weekdaysShort : "Zo_Ma_Di_Wo_Do_Vr_Za".split("_")
        };

        return (
            <div>
                <InfiniteCalendar
                    ...
                    locale={locale}
                />
            </div>
```

_(I've changed "August" to "testing" and "Aug" to "tst")_

The labels of the days of the week get translated correctly. However, the months do not:

![screen shot 2016-08-12 at 15 16 16](https://cloud.githubusercontent.com/assets/331833/17615321/050625d2-60a0-11e6-8718-a4c8f44ae6c2.png)

(Note that the multiple dates come from my own fork with support for this, note that I didn't change anything related to locales)

When debugging your library I found that the locale is set correctly to `moment.js`. E.g.:

_index.js_

```
updateLocale(locale) {
    locale = this.getLocale(locale);
    moment.updateLocale(locale.name, locale);
    moment.locale(locale.name);

    console.log('logging date format at init');
    const date = moment();
    console.log(date.format('MMM'));
}
```

will correctly output `tst`

But inside _List/index.js_:

```
renderMonth = ({index, isScrolling}) => {
    let {...} = this.props;
    let {date, rows} = this.memoize(months[index]);

    console.log('logging date format at list render month');
    const date2 = moment();
    console.log(date.format('MMM'));
    console.log(date2.format('MMM'));
```

Date2 (the newly created moment) is correct: `tst`. However, the original `date` coming from `this.memoize` will be English (`Aug`).

I believe this is because this date comes from another instance of moment.js inside util.js

_utils.js_

```
export function getMonth(monthDate) {
    let rows = {};
    let daysInMonth = monthDate.daysInMonth();
    let year = monthDate.year();
    let month = monthDate.month();

    let week, date, lastWeekVal;
    let weekIndex = -1;

    for (let i = 0; i < daysInMonth; i++) {
        date = moment(new Date(year,month,i+1)); //this moment.js does not have locale applied
                ....
```

But I have no apparent solution for now. What do you think is the cause of these different instances of moment? Do we need to re-apply locale every time moment.js is imported?

**Comment by clauderic** [comment]
Hmm. What do you mean by different instances of moment.js? AFAIK they are all one and the same instance, a simple look at the generated output from Webpack can confirm this.

**Comment by tommedema** [comment]
@clauderic I don't mean that webpack injects the module multiple times into the bundle. That's not the problem.

When I set the locale, the locale is activated by the instance of moment.js inside `index.js` and `List/index.js` etc., but not by the moment instance used at `utils.js`.

Can I provide any more information?

**Comment by clauderic** [comment]
Hey @tommedema, just shipped version `2.0.0`, which replaces `moment.js` with `date-fns`. This should no longer be an issue. 

Let me know if you run into any other issues with version 2.0
